### ISSUE TITLE
Deploy Keys

### ISSUE THREAD

**Issue by yfeldblum** [applicationsonline/librarian#2]
Permit:

```
# Cheffile
cookbook 'my-cookbook',
  :git => 'https://github.com/my-private/cookbook-repo',
  :key => 'deploy-keys/my-private/cookbook-repo.key'

# Gemfile
gem 'my-gem',
  :git => 'https://github.com/my-private/gem-repo',
  :key => 'deploy-keys/my-private/gem-repo.key'
```

With the directory `deploy-keys` in this example being a directory relative to the specfile.

`Librarian::Source::Git::Repository` would construct a `git-ssh` shell script and set the `GIT_SSH` environment variable for the `git` process for these specs.

This is useful if one has a _private_ dependency (gem/cookbook/etc) in a private but _separate_ git repository. One is then able to use the `git` source and point to the external repository, without being required to vendor the external private dependency into the current project. It is best practice to split distinct software components into distinct repositories, and let Librarian manage resolving all of the dependencies; and this feature would facilitate that.

One problem is: constructing a git-ssh server on the local machine such that deploy keys may be tested.

**Comment by smdern** [comment]
Try using the git url instead of https

**Comment by yfeldblum** [comment]
Correct, the example should have used the `git@github.com:user/repo.git` form of the URL.

Nevertheless the problem remains: "constructing a git-ssh server on the local machine such that deploy keys may be tested" without needing to be root to run the tests.
